RadClass/analysis.py

```diff
@@ -0,0 +1,205 @@
+        
+        NOTE: Returned data is normalized for live times (1s measurement
+        period less dead time). i.e. The returned data is a count rate at
+        each bin.
+
+        Return: 1D numpy array for integration interval
+        '''
+
+        # extract requisite data rows
+        data_matrix = self.processor.data_slice(self.datapath, rows)
+
+        # normalize by live times to produce count rate data
+        # processor.live can be indexed by appropriate timestamps but
+        # data_matrix only has indices for rows.
+        for idx, row in enumerate(rows):
+            dead_time = self.processor.live[row]
+            data_matrix[idx] = data_matrix[idx] / dead_time
+
+        # old, more inefficient way of summing
+        #total = np.zeros_like(data_matrix[0])
+        #for row in data_matrix:
+        #    total += row
+
+        # utilizes numpy architecture to sum data
+        total = np.sum(data_matrix, axis = 0)
+
+        return total
+
+    def collect_rows(self):
+        '''
+        Uses the integration duration to collect all rows needed for analysis.
+        Uses numpy.arange() to enumerate row indices from start to end.
+
+        Return: list of rows for data indexing
+        '''
+
+        # collect start index from tracked timestamp
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+
+        # NOTE: this behavior is currently disabled by the if-statement in
+        # march() removing the 'or' portion with allow this to work
+        #
+        # if the final portion of the file is smaller than a full integration
+        # interval, only what is left is collected for this analysis
+        if start_i + self.integration >= len(self.processor.timestamps):
+            end_i = len(self.processor.timestamps) - 1
+        else:
+            end_i = start_i + self.integration
+
+        # enumerate number of rows to integrate exclusive of the endpoint
+        rows = np.arange(start_i, end_i)
+        return rows
+
+    def march(self):
+        '''
+        Advance a step using stride time to find the new working integration
+        interval. Also keeps track of the EOF state but checking for
+        out-of-index situations.
+
+        Returns: boolean for EOF
+        '''
+
+        # find the working integration interval starting index and advance stride
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+        start_i = start_i[0]
+        new_i = start_i + self.stride
+
+        # stop analysis if EOF reached
+        # NOTE: stops prematurely, only does analysis for windows of full integration time
+        if new_i >= len(self.processor.timestamps) or (new_i + self.integration) >= len(self.processor.timestamps):
+            return False
+        else:
+            # update working integration interval timestep
+            self.working_time = self.processor.timestamps[new_i]
+            return True
```

Comment:
I have a mild preference for simpler conditionals and single return points in cases like this:
(also PEP8 line length)
```suggestion
        running = True
        if new_i >= (len(self.processor.timestamps) or 
                           (new_i + self.integration) >= len(self.processor.timestamps)):
            running = False

        if running:  
             # update working integration interval timestep
            self.working_time = self.processor.timestamps[new_i]

        return running
```